projects/plugins/jetpack/modules/masterbar/admin-menu/class-admin-menu.php

```diff
@@ -104,6 +104,54 @@ public function add_my_home_menu() {
 		$this->update_menu( 'index.php', 'https://wordpress.com/home/' . $this->domain, __( 'My Home', 'jetpack' ), 'manage_options', 'dashicons-admin-home' );
 	}
 
+	/**
+	 * Adds upsell nudge as a menu.
+	 *
+	 * @param object $nudge The $nudge object containing the content, CTA, link and tracks.
+	 */
+	public function add_upsell_nudge( $nudge ) {
+		$message = '
+<div class="upsell_banner">
+	<div class="banner__info">
+		<div class="banner__title">%1$s</div>
+	</div>
+	<div class="banner__action">
+		<button type="button" class="button">%2$s</button>
+	</div>
+</div>';
+
+		$message = sprintf(
+			$message,
+			$nudge['content'],
+			$nudge['cta']
```

Comment:
Do we need to escape / wp_kses some of those strings?

Reply:
Very good, idea. Although current upsell nudges do not contain HTML, better safe than sorry!

Thanks